projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php

```diff
@@ -633,4 +633,53 @@ public function test_from_story_block_from_post_id_is_correct_array_videopress_w
 		$expected_poster_url = str_replace( 'mp4', 'jpg', $post_info['img_urls'][1] );
 		$this->assertEquals( $expected_poster_url, $images[1]['src'] );
 	}
+
+	/**
```

Comment:
Thanks for adding tests! ❤️ 

This test file has been around for quite some time, and is not very optimized. As you're adding a new test, I think it would make sense to switch to using a data provider method so you do not have to repeat things too much. Here is how you could do it:

```diff
diff --git a/projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php b/projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php
index f85335b417..eeea9e841e 100644
--- a/projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php
+++ b/projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php
@@ -635,51 +635,52 @@ class WP_Test_Jetpack_PostImages extends WP_UnitTestCase {
 	}
 
 	/**
-	 * Test if the array extracted is empty in case post_id is null.
+	 * Test if the array extracted is empty in case post_id is a string.
 	 *
 	 * @covers Jetpack_PostImages::from_gravatar
-	 * @since 11.3
-	 */
-	public function test_from_gravatar_returns_empty_array_if_post_id_is_null() {
-		$post_id = null;
-		$images  = Jetpack_PostImages::from_gravatar( $post_id );
-
-		$this->assertEmpty( $images );
-	}
-
-	/**
-	 * Test if the array extracted is empty in case post_id is an integer.
+	 * @dataProvider provider_gravatar
 	 *
-	 * @covers Jetpack_PostImages::from_gravatar
-	 * @since 11.3
+	 * @since $$next-version$$
+	 *
+	 * @param int|string|null $post_id  The post ID.
+	 * @param array           $expected The expected array.
 	 */
-	public function test_from_gravatar_returns_empty_array_if_post_id_is_integer() {
-		$post_id = 5;
-		$images  = Jetpack_PostImages::from_gravatar( $post_id );
-
-		$this->assertEmpty( $images );
+	public function test_from_gravatar( $post_id, $expected ) {
+		$image_details = Jetpack_PostImages::from_gravatar( $post_id );
+		$this->assertEquals( $expected, $image_details );
 	}
 
 	/**
-	 * Test if the array extracted has a valid image when sending a valid post.
+	 * Test data for our tests for Jetpack_PostImages::from_gravatar.
 	 *
-	 * @covers Jetpack_PostImages::from_gravatar
-	 * @since 11.3
+	 * @retun array
 	 */
-	public function test_from_gravatar_returns_valid_image() {
-
-		$post_id = $this->factory->post->create();
+	public function provider_gravatar() {
+		$valid_post_id = $this->factory->post->create();
 
-		$images = Jetpack_PostImages::from_gravatar( $post_id );
-
-		$this->assertCount( 1, $images );
-		$this->assertEquals( 'image', $images[0]['type'] );
-		$this->assertEquals( 'gravatar', $images[0]['from'] );
-		$this->assertStringContainsString( 'gravatar.com/avatar', $images[0]['src'] );
-		$this->assertEquals( 96, $images[0]['src_width'] );
-		$this->assertEquals( 96, $images[0]['src_height'] );
-		$this->assertNotEmpty( $images[0]['href'] );
-		$this->assertSame( '', $images[0]['alt_text'] );
+		return array(
+			'invalid (null) post id'                  => array(
+				null,
+				array(),
+			),
+			'post id does not match an existing post' => array(
+				5,
+				array(),
+			),
+			'valid post'                              => array(
+				$valid_post_id,
+				array(
+					array(
+						'type'       => 'image',
+						'from'       => 'gravatar',
+						'src'        => 'http://1.gravatar.com/avatar/?s=96&d=mm&r=g',
+						'href'       => 'http://example.org/?p=4',
+						'src_width'  => 96,
+						'src_height' => 96,
+						'alt_text'   => '',
+					),
+				),
+			),
+		);
 	}
-
 } // end class
```

Down the road, it may be a good task to refactor the whole test file to rely on data providers more. That's for another PR though :)

Reply:
Thanks a lot for the suggestion. I pushed a commit following the suggestion